File: 5.1C-B/app/src/main/java/com/rorycd/istream/MainActivity.java
package com.rorycd.istream;

import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;

import com.rorycd.istream.data.UserRepository;
import com.rorycd.istream.ui.HomeFragment;
import com.rorycd.istream.ui.LoginFragment;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // On the first OnCreate
        if (savedInstanceState == null) {
            UserRepository repo = new UserRepository(this);

            Fragment initial = repo.isLoggedIn() ? HomeFragment.newInstance() : LoginFragment.newInstance();

            changeToFragment(initial);
        }

        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

    }

    public void changeToFragment(Fragment fragment) {
        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.main, fragment).addToBackStack(null).commit();
    }
}

File: 8.2HD/app/src/main/java/com/rorycd/bowerbird/MainActivity.kt
package com.rorycd.bowerbird

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.activity.enableEdgeToEdge
import com.rorycd.bowerbird.ui.main.MainAppScreen
import com.rorycd.bowerbird.ui.theme.BowerbirdTheme
import dagger.hilt.android.AndroidEntryPoint

@AndroidEntryPoint
class MainActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContent {
            BowerbirdTheme {
                MainAppScreen()
            }
        }
    }
}
